ntge-core/src/message/encryptor.rs

```diff
@@ -0,0 +1,88 @@
+use chrono::prelude::*;
+use hkdf::Hkdf;
+use hmac::{Hmac, Mac};
+use rand::rngs::OsRng;
+use rand::RngCore;
+use secrecy::ExposeSecret;
+use sha2::Sha256;
+use x25519_dalek::PublicKey;
+
+use crate::{aead, message, x25519::FileKey};
+
+#[derive(Debug)]
+pub struct Encryptor {
+    pub recipent_public_keys: Vec<PublicKey>,
+}
+
+impl Encryptor {
+    pub fn new(recipent_public_keys: Vec<PublicKey>) -> Self {
+        Encryptor {
+            recipent_public_keys,
+        }
+    }
+}
+
+#[allow(dead_code)]
+impl Encryptor {
+    pub fn encrypt(&self, plaintext: &[u8]) -> message::Message {
+        // 1. create new file key
+        let file_key = FileKey::new();
+        // 2. wrap file key for recipients
+        let recipient_headers: Vec<_> = self
+            .recipent_public_keys
+            .iter()
+            .map(|public_key| file_key.wrap(&public_key))
+            .collect();
+        let meta = message::MessageMeta {
+            timestamp: Some(Utc::now().to_string()),
+        };
+        // 3. TODO: calculate HMAC for recipient_headers + meta
+        let mac = Encryptor::calculate_mac(&recipient_headers, &meta, &file_key);
+        // 4. create payload
+        // prepare nonce
+        let mut nonce = [0; 16];
+        OsRng.fill_bytes(&mut nonce);
+        // create payload key
+        let mut payload_key = [0; 32];
+        Hkdf::<Sha256>::new(Some(&nonce), file_key.0.expose_secret())
+            .expand(message::PAYLOAD_KEY_LABEL, &mut payload_key)
+            .expect("payload_key is the correct length");
+        let ciphertext = aead::aead_encrypt(&payload_key, &plaintext);
+        let nonce = nonce.to_vec();
+        let payload = message::MessagePayload { nonce, ciphertext };
+        // 5. construct message
+        let mac = message::MessageMac { mac: mac.to_vec() };
+        message::Message {
+            recipient_headers,
+            meta,
+            mac,
+            payload,
+        }
+    }
+}
+
+impl Encryptor {
+    pub(crate) fn calculate_mac(
```

Comment:
Since this fn is only used within crate `message` how about
```suggestion
    pub(super) fn calculate_mac(
```